### PR TITLE
DEPR: Deprecating .clip_lower and .clip_upper

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] xref #18262
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

**Review by jorisvandenbossche — APPROVED** [review]

File: doc/source/whatsnew/v0.24.0.rst [hunk 1]
```diff
@@ -1113,6 +1113,8 @@ Deprecations
 - :func:`pandas.types.is_datetimetz` is deprecated in favor of `pandas.types.is_datetime64tz` (:issue:`23917`)
 - Creating a :class:`TimedeltaIndex` or :class:`DatetimeIndex` by passing range arguments `start`, `end`, and `periods` is deprecated in favor of :func:`timedelta_range` and :func:`date_range` (:issue:`23919`)
 - Passing a string alias like ``'datetime64[ns, UTC]'`` as the `unit` parameter to :class:`DatetimeTZDtype` is deprecated. Use :class:`DatetimeTZDtype.construct_from_string` instead (:issue:`23990`).
+- :meth:`Series.clip_lower`, :meth:`Series.clip_upper`, :meth:`DataFrame.clip_lower` and :meth:`DataFrame.clip_upper` are deprecated and will be removed in a future version.
+  Use ``Series.clip(lower=threshold)``, ``Series.clip(upper=threshold)`` and the equivalent ``DataFrame`` methods (:issue:`24203`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.rst]: is this on another line?

**Review by jreback — APPROVED** [review]
looks fine. one small comment. pls add to #6581 as well

**Comment by jreback** [comment]
and pls merge master

**Comment by jreback** [comment]
lgtm. merge on green.

**Pull request merged**